src/model/log_prob.jl

```diff
@@ -0,0 +1,443 @@
+
+
+"""
+Creates a vectorized version of the star logpdf as a function of
+unconstrained params.
+    star_params = [lnr, lnc1, ..., lnc4, ra, dec]
+Args:
+  - images: Vector of TiledImage types (data for log_likelihood)
+  - active_pixels: Vector of ActivePixels on which the log_likelihood is based
+  - ea: ElboArgs book keeping argument
+Returns:
+  - star_logpdf  : unnormalized logpdf function handle that takes in a flat,
+                   unconstrained array as parameter
+  - star_logprior: star param log prior function handle that takes in same
+                   flat, unconstrained array as parameter
+"""
+function make_star_logpdf(images::Vector{TiledImage},
+                          active_pixels::Vector{ActivePixel},
+                          S::Int64,
+                          N::Int64,
+                          vp::VariationalParams{Float64},
+                          tile_source_map::Vector{Matrix{Vector{Int}}},
+                          patches::Matrix{SkyPatch},
+                          active_sources::Vector{Int})
+
+    # define star prior log probability density function
+    prior    = construct_prior()
+    subprior = prior.star
+
+    function star_logprior(state::Vector{Float64})
+        brightness, colors, u = state[1], state[2:5], state[6:end]
+        return color_logprior(brightness, colors, prior, true)
+    end
+
+    # define star log joint probability density function
+    function star_logpdf(state::Vector{Float64})
+        ll_prior = star_logprior(state)
+        brightness, colors, position = state[1], state[2:5], state[6:end]
+        dummy_gal_shape = [.1, .1, .1, .1]
+        ll_like  = state_log_likelihood(true, brightness, colors, position,
+                                        dummy_gal_shape, images,
+                                        active_pixels, 
+                                        patches,
+                                        active_sources, vp,
+                                        tile_source_map, S, N)
+        return ll_like + ll_prior
+    end
+    return star_logpdf, star_logprior
+end
+
+
+#"""
+#Creates a vectorized version of a galaxy logpdf as a function of
+#unconstrained params
+#    gal_params = [lnr, lnc1, ..., lnc4, ra, dec, shp1, shp2, shp3, shp4]
+#where the flux and position params are the same as the star case, and the
+#shape params are unconstrained versions of (gdev, gaxis, gangle, gscale)
+#Args:
+#  - images: Vector of TiledImage types (data for log_likelihood)
+#  - active_pixels: Vector of ActivePixels on which the log_likelihood is based
+#  - ea: ElboArgs book keeping argument
+#Returns:
+#  - gal_logpdf  : unnormalized logpdf function handle that takes in a flat,
+#                 unconstrained array as parameter
+#  - gal_logprior: star param log prior function handle that takes in same
+#                 flat, unconstrained array as parameter
+#"""
+#function make_galaxy_logpdf(images::Vector{TiledImage},
```

Comment:
I forgot about this, I will incorporate this function and add the corresponding test.